mlflow/sklearn/__init__.py

```diff
@@ -254,14 +257,15 @@ def save_model(
         serialization_format=serialization_format,
     )
 
-    # `PyFuncModel` only works for sklearn models that define `predict()`.
-    if hasattr(sk_model, "predict"):
+    # `PyFuncModel` only works for sklearn models that define a predict function
+    if hasattr(sk_model, pyfunc_predict_fn):
```

Comment:
Can we log a warning if the specified predict function isn't present on the model indicating that the pyfunc model flavor won't be added to the resulting model?

Reply:
Great suggestion! Done!

But maybe another wrench here... 😅. The straightforward way to add a warning log, like I did, will mean that warnings go off for what is possibly legitimate code today (though I'm not familiar with scikit-learn models without a predict function). This code structure makes me think that perhaps what we want is something like:

```
def save_model(..., pyfunc_predict_fn=None):
  ...
  # When
  if not pyfunc_predict_fn and hasattr(sk_model, 'predict'):
    pyfunc_predict_fn = 'predict'

  if pyfunc_predict_fn:
    if not hasattr(sk_model, pyfunc_predict_fn):
      _logger.warning(f'Model was missing {pyfunct_predict_fn} function!')
    else:
      
```

It looks uglier to me (perhaps better way to write it?) But would avoid warnings going off for what is today legitimate code. I'm not familiar enough with scikit-learn models w/o `predict` methods to know if this is worth doing -- so feel free to merge if not. But let me know if it is and I'm happy to comply :)

Reply:
I think that adding warning for the existing case(s) where users are logging models that don't define `predict()` is just fine :D. We can always revisit this if necessary. Thanks for considering this!